### PR TITLE
APR-2601 - UKPRN allow list page

### PR DESCRIPTION
**Pull request by martin-lush**

### CODE REVIEW

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.AdminService.Web/Validators/Roatp/AllowedProviders/AddUkprnToAllowProvidersListValidator.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    {
+                        if (!int.TryParse(ukprnInput, out var ukprn))
+                        {
+                            context.AddFailure("Enter a valid UKPRN");
+                        }
+                        else if (ukprn < 10000000 || ukprn > 19999999)
+                        {
+                            context.AddFailure("Enter a valid UKPRN using 8 numbers");
+                        }
+                        else if (await _applyApiClient.GetAllowedProviderDetails(ukprn) != null)
```
Comment (chrisfoster76) [inline src/SFA.DAS.AdminService.Web/Validators/Roatp/AllowedProviders/AddUkprnToAllowProvidersListValidator.cs:28]: This code will run synchronously

**Review by chrisfoster76 — COMMENTED** [review]
Just a comment about async await being ignored in this context

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/RoatpAllowedProvidersController.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            return View("~/Views/Roatp/AllowedProviders/List.cshtml", model);
+        }
+
+        [HttpPost("/Roatp/AllowedProviders")]
+        public async Task<IActionResult> AddUkprn(AddUkprnToAllowedProvidersListViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                model.AllowedProviders = await _applyApiClient.GetAllowedProvidersList(model.SortColumn, model.SortOrder);
+                return View("~/Views/Roatp/AllowedProviders/List.cshtml", model);
```
Comment (chrisfoster76) [inline src/SFA.DAS.AdminService.Web/Controllers/Roatp/RoatpAllowedProvidersController.cs:43]: Rendering a view off a Post will break the back button if subsequently used to return to the page

Comment (martin-lush) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/RoatpAllowedProvidersController.cs]: Have now implemented PRG

File: src/SFA.DAS.AdminService.Web/Validators/Roatp/AllowedProviders/AddUkprnToAllowProvidersListValidator.cs (hunk 1 above)
Comment (martin-lush) [reply src/SFA.DAS.AdminService.Web/Validators/Roatp/AllowedProviders/AddUkprnToAllowProvidersListValidator.cs:28]: In the MVC pipeline it does but the only thing to do would be to use a GetAwaiter().GetResult() which I believe will be done under the hood.

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — DISMISSED** [review]

**Pull request merged**